pkg/kubelet/cm/deviceplugin/manager.go

```diff
@@ -497,6 +499,16 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 		return nil, fmt.Errorf("can't allocate unregistered device %v", resource)
 	}
 	devices = sets.NewString()
+	// Gets devices allocated to init containers and allocates from that list first.
+	for _, initCont := range initContainers {
+		for device := range m.podDevices.containerDevices(podUID, initCont, resource) {
```

Comment:
Thus, there is possibility that multiple regular containers reuse a same device, which is not allowed.
Maybe we need a flag to mark whether a device has been reused?

Reply:
Good catch! I changed to track devices that we may reuse from init containers and subtract from this list when allocating for normal containers.

Reply:
Yeah，I see :) so I guess the main issue about the option now is if it is supported in certain plugins that a device be allocated multi times before containers requiring it actually running.